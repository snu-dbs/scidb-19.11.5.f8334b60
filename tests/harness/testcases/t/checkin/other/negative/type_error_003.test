--setup
create array mult_1 <a:double> [x=0:2,1,0, y=0:2,1,0]
load(mult_1,'${TEST_DATA_DIR}/mult1.txt')
scan(mult_1)  -- Inserted by scan_doctor

--test
--error --code=scidb::SCIDB_SE_QPROC::SCIDB_LE_FUNCTION_NOT_FOUND "apply(mult_1, Z, x / 'a string')"

# <string> * <int> now works though!
apply(mult_1, Z, 'Ho! ' * x)
apply(mult_1, Z, (3-x) * 'Nyah! ')
build(<s:string>[i=-2:3], 'What would Bender do? ' * i)

--cleanup
remove (mult_1)
