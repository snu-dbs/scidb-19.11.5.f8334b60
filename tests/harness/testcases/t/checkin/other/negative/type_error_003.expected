Query was executed successfully

Query was executed successfully

{x,y} a
{0,0} 1
{0,1} 2
{0,2} 3
{1,0} 4
{1,1} 5
{1,2} 6
{2,0} 7
{2,1} 8
{2,2} 9

[An error expected at this place for the query "apply(mult_1, Z, x / 'a string')". And it failed with error code = scidb::SCIDB_SE_QPROC::SCIDB_LE_FUNCTION_NOT_FOUND. Expected error code = scidb::SCIDB_SE_QPROC::SCIDB_LE_FUNCTION_NOT_FOUND.]

{x,y} a,Z
{0,0} 1,''
{0,1} 2,''
{0,2} 3,''
{1,0} 4,'Ho! '
{1,1} 5,'Ho! '
{1,2} 6,'Ho! '
{2,0} 7,'Ho! Ho! '
{2,1} 8,'Ho! Ho! '
{2,2} 9,'Ho! Ho! '

{x,y} a,Z
{0,0} 1,'Nyah! Nyah! Nyah! '
{0,1} 2,'Nyah! Nyah! Nyah! '
{0,2} 3,'Nyah! Nyah! Nyah! '
{1,0} 4,'Nyah! Nyah! '
{1,1} 5,'Nyah! Nyah! '
{1,2} 6,'Nyah! Nyah! '
{2,0} 7,'Nyah! '
{2,1} 8,'Nyah! '
{2,2} 9,'Nyah! '

{i} s
{-2} ''
{-1} ''
{0} ''
{1} 'What would Bender do? '
{2} 'What would Bender do? What would Bender do? '
{3} 'What would Bender do? What would Bender do? What would Bender do? '

Query was executed successfully

